grpc/federation/cel/plugin.go

```diff
@@ -184,8 +192,27 @@ func (i *CELPluginInstance) ValidatePlugin(ctx context.Context) error {
 	return nil
 }
 
+func (i *CELPluginInstance) write(cmd []byte) error {
+	if i.closed {
+		return errors.New("grpc-federation: plugin has already been closed")
+	}
+
+	writeCh := make(chan error)
+	go func() {
+		_, err := i.stdin.Write(cmd)
+		writeCh <- err
+	}()
+	select {
+	case err := <-i.instanceModErrCh:
```

Comment:
What if CELPluginInstance#write is called before `p.wasmRuntime.InstantiateModule` finishes running? 🤔 Should we somehow wait until the instantiation is completed...?

Reply:
Ah, no. Sorry I was wrong...

Reply:
Ahh wait, but what if `p.wasmRuntime.InstantiateModule` enqueues nil to `instanceModErrCh` and `writeCh` receives an error from `i.stdin.Write(cmd)`? In that case, don't we end up missing the error from `i.stdin.Write(cmd)`...?

Reply:
`instanceModErrCh` returns an error when the module fails to start correctly (e.g., if `context.Context` has already been canceled ). `InstantiateModule` will continue to run until the plugin is no longer in use. So, we must call `(*CELPluginInstance).Close()` to send an exit signal to finish the wasm module. For this reason, `nil` is usually not returned from `instanceModErrCh`